src/components/settings/ImportAllDialog/index.tsx

```diff
@@ -0,0 +1,237 @@
+import { SETTINGS_EVENTS, trackEvent } from '@/services/analytics'
+
+const AcceptedMimeTypes = {
+  'application/json': ['.json'],
+}
+
+const GreyFileIcon = () => (
+  <SvgIcon component={FileIcon} inheritViewBox fontSize="small" color="border" sx={{ fill: 'none' }} />
+)
+
+const ImportAllDialog = ({ handleClose }: { handleClose: () => void }): ReactElement => {
+  const [jsonData, setJsonData] = useState<string>()
+  const [fileName, setFileName] = useState<string>()
+  const [error, setError] = useState<string>()
+
+  // Parse the jsonData whenever it changes
+  const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount } = useGlobalImportJsonParser(jsonData)
+
+  const dispatch = useAppDispatch()
+
+  const onDrop = (acceptedFiles: File[]) => {
+    if (acceptedFiles.length === 0) {
+      return
+    }
+    const file = acceptedFiles[0]
+    const reader = new FileReader()
+    reader.onload = function (evt) {
+      if (!evt.target) {
+        return
+      }
+      if (typeof evt.target.result !== 'string') {
+        return
+      }
+      setJsonData(evt.target.result)
+      setFileName(file.name)
+    }
+    reader.readAsText(file)
+  }
+
+  const onRemove: MouseEventHandler = (event) => {
+    setJsonData(undefined)
+    setFileName(undefined)
+    event.preventDefault()
+    event.stopPropagation()
+  }
+
+  const { getRootProps, getInputProps, isDragActive, isDragAccept, isDragReject } = useDropzone({
+    maxFiles: 1,
+    onDrop,
+    accept: AcceptedMimeTypes,
+  })
+
+  const handleImport = () => {
+    if (!addressBook && !addedSafes) {
+      return
+    }
+
+    if (addressBook) {
+      dispatch(addressBookSlice.actions.setAddressBook(addressBook))
+
+      trackEvent({
+        ...SETTINGS_EVENTS.DATA.IMPORT_ADDRESS_BOOK,
+        label: addressBookEntriesCount,
+      })
+    }
+
+    if (addedSafes) {
+      dispatch(addedSafesSlice.actions.setAddedSafes(addedSafes))
+
+      trackEvent({
+        ...SETTINGS_EVENTS.DATA.IMPORT_ADDED_SAFES,
+        label: addedSafesCount,
+      })
+    }
+
+    dispatch(
+      showNotification({
+        variant: 'success',
+        groupKey: 'global-import-success',
+        message: 'Successfully imported data',
+        detailedMessage: `${addedSafesCount > 0 ? `${addedSafesCount} Safes were added.` : ''}${
+          addressBookEntriesCount > 0 ? `\n${addressBookEntriesCount} addresses were added to your address book.` : ''
+        }`,
+      }),
+    )
+
+    handleClose()
+  }
+
+  return (
+    <ModalDialog open onClose={handleClose} dialogTitle="Data import" hideChainIndicator>
+      <DialogContent>
+        <div>
+          {fileName ? (
+            <Grid container direction="column" gap={1}>
+              <Grid container gap={1} display="flex" alignItems="center">
+                <Grid item xs={1}>
+                  <SvgIcon component={FileIcon} inheritViewBox fontSize="small" color="primary" sx={{ fill: 'none' }} />
```

Comment:
You can leverage `GreyFileIcon` here if you allow it to receive the `color` prop.